### PR TITLE
Disallow running projects on Databricks with a local tracking URI

### PR DESCRIPTION
**Pull request by smurching**
Moves existing validation that prevented running a project with a local tracking URI earlier in the mlflow.projects.run logic - before we attempted to default the tracking URI to "databricks" when executing a project on Databricks, but only validated the tracking URI after we'd already created a run against the (potentially local FileStore) tracking server. This PR removes the logic for defaulting the tracking URI to "databricks" in favor of requiring users to explicitly specify a "databricks" or "http://" tracking URI

### CODE REVIEW

File: mlflow/projects/databricks.py [hunk 1]
```diff
@@ -229,8 +232,6 @@ def jobs_runs_get(self, databricks_run_id):
 
 
 def _get_tracking_uri_for_run():
-    if not tracking.utils.is_tracking_uri_set():
```
Comment (smurching) [inline mlflow/projects/databricks.py]: We remove the old behavior here, which attempted to default the tracking URI to "databricks" if it was not set. Unfortunately, this logic was called only after we'd already created a local run, so the Databricks Job would attempt to fetch a run with the UUID of the local run from the Databricks tracking server.

File: mlflow/projects/__init__.py [hunk 2]
```diff
@@ -38,6 +39,9 @@ def _run(uri, entry_point="main", version=None, parameters=None, experiment_id=N
     Helper that delegates to the project-running method corresponding to the passed-in mode.
     Returns a ``SubmittedRun`` corresponding to the project run.
     """
+    if mode == "databricks":
+        mlflow.projects.databricks.before_run_validations(mlflow.get_tracking_uri(), cluster_spec)
+
     exp_id = experiment_id or _get_experiment_id()
```
Comment (dbczumar) [inline mlflow/projects/__init__.py]: How do you feel about moving the project run creation logic into a separate method: `_create_project_run()`? We can then move the new `mode` check into the condition on line 67 and call `_create_project_run()` in both the if/elif branches.

**Review by tomasatdatabricks — APPROVED** [review]
LGTM. Looks good thanks Sid.

My only comment / question is whether we need to check authentication. It's not really that important but it seems like it's an extra code which does not guarantee that the authentication is correct and is not really needed so I would consider removing it altogether.

**Comment by smurching** [comment]
Discussed offline with @dbczumar and @tomasatdatabricks, we'll defer refactoring the run-creation logic to later. Went ahead and removed the auth-checking logic, agree that it's not super useful as is (it just checks that a Databricks CLI profile exists, but not that the credentials in the profile are valid). I'd like to add back the check in a follow-up, making an API request (like listing available spark versions) to verify the credentials are valid. This check is mainly useful in the case where the project is running on Databricks against a non-Databricks tracking server - if the tracking server is a Databricks server, we'll implicitly verify the auth is valid when we make an API request to create a Run.

Thanks both of you for the reviews!

**Comment by smurching** [comment]
Merging, thanks again @tomasatdatabricks @dbczumar!

**Pull request merged**